### ISSUE TITLE
list() does not respect max: 1

### ISSUE THREAD

**Issue by schnittchen** [pragdave/pollution#6]
(continuing https://github.com/pragdave/quixir/issues/9)

```
alias Pollution.Generator.List
{_,list}=Enum.reduce(1..10, {List.create(of: Pollution.VG.bool(), max: 1),[]}, fn _, {state,list} -> {v,state}= List.next_value(state, []); {state, list ++ [v]} end); list
```

produces the same problem, some levels deeper.

**Comment by schnittchen** [comment]
And I think this is the problem:

```
iex(44)> Enum.map(1..0, fn _ -> nil end)
[nil, nil]
```

Here https://github.com/pragdave/pollution/blob/master/lib/pollution/generators/list.ex#L83, when len equals 0.

**Comment by schnittchen** [comment]
Closed by #7
